pybsn/__init__.py

```diff
@@ -26,38 +26,138 @@
+        """
         return self._connection.post(self._path, data)
 
     def put(self, data):
+        """ Replaces (PUT) the given data in BigDB at the path identified by this node.
+
+        :param data: JSON serializable data to post, often a dictionary.
+            Note that the data provided here is passed to BigDB as-is; i.e., use hyphens.
+        """
         return self._connection.put(self._path, data)
 
     def patch(self, data):
+        """ Updates (PATCH) the given data in BigDB at the path identified by this node.
+
+        :param data: JSON serializable data to post, often a dictionary.
+            Note that the data provided here is passed to BigDB as-is; i.e., use hyphens.
+        """
         return self._connection.patch(self._path, data)
 
     def delete(self):
+        """ Delete the data stored in BigDB at the path identified by this node. """
         return self._connection.delete(self._path)
 
     def schema(self):
+        """ Retrieve the schema for BigDB at the path identified by this node. """
         return self._connection.schema(self._path)
 
     def rpc(self, data=None):
+        """ Invoke the BigDB RPC endpoint identified by this node.
+
+         :param data to provide as RPC input to BigDB.
+            Note that the data provided here is passed to BigDB as-is; i.e., use hyphens.
+         :return the output data returned by the RPC endpoint.
+        """
         return self._connection.rpc(self._path, data)
 
+    def match(self, **kwargs):
+        """ Adds exact match predicates to the path represented by the current Node. Returns
+        a Node representing the new path.
+
+        Supply the child element(s) to match on as keyword parameters:
+        * The name of the parameter indicates the child element to match on
+           (hyphens converted to underscores)
+        * The value of the parameter indicates the desired value
+
+        E.g.,
+        node.match(mac_address="01:02:03:04:05:05:06")
+        translates into the BigDB Path
+
+        [...]/node[mac-address="14:18:77:96:8b:d6"]
+        """
+        for k, v in kwargs.items():
+            self = self.filter("%s=$x" % k.replace('_', '-'), x=v)
+        return self
+
     def filter(self, template, *args, **kwargs):
+        """ Adds a predicate to the path represented by the current Node. Returns
+        a Node representing the new path.
+
+        :param :template the predicate to add. Any template variables (e.g., $x)
+               will be replaced with values retrieved from the keyword arguments
+
+        E.g., to retrieve all BCF segments with a member ID smaller than 1000, you
+        might do:
+        root.applications.bcf.tenant.segment.filter("member-vlan<$max", max=1000).get()
+
+        Here,
+        segment.filter("member-vlan<$max", max=1000)
+        translates into the BigDB path:
+        [...]/segment[member-vlan<1000]
```

Comment:
Square bracket `[]` is valid character representing BigDB predicate, so there might be lesser chance of confusion using something else for abbreviation? e.g., `~/segment`, `.../segment`.